#!/usr/bin/env python
PACKAGE = "robocup_action_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('MaxStepX', double_t, 0, "maximum foward transition along X(meters)", 0.080, 0.001, 0.04)
gen.add('MaxStepY', double_t, 0, "absolute maximum lateral transition along Y(meters)", 0.160, 0.101, 0.14)
gen.add('MaxStepTheta', double_t, 0, "absolute maximum rotation around Z(radians)", 0.524, 0.001, 0.349)
gen.add('MaxStepFrequency', double_t, 0, "maximum step frequency(normalized, unit-less)", 1.0, 0.0, 1.0)
gen.add('StepHeight', double_t, 0, "peak foot elevation along Z(meter)", 0.040, 0.005, 0.020)
gen.add('TorsoWx', double_t, 0, "peak Torso rotation along X(radian)", 0.000, -0.122, 0.122)
gen.add('TorsoWy', double_t, 0, "peak Torso rotation along Y(radian)", 0.000, -0.122, 0.122)

gen.add('UseSmartStifness', bool_t, 0, "use smart stifness or not", True)
gen.add('KillAutonomousLife', bool_t, 0, "kill autonomous life or not", True)
gen.add('UseWholeBodyBalancer', bool_t, 0, "use whole body balancer or not", True)


exit(gen.generate(PACKAGE, "naoqi_robocup_walker", "naoqi_robocup_walker_dynamic_params"))
